### PR TITLE
Lua selection changes

### PR DESCRIPTION
**Pull request by wpferguson**
Added name field to darktable.gui.select.register_selection() so that registered selections (buttons in the select images) can be found and manipulated.

Added darktable.gui.select.destroy_selection() to remove a lua registered selection.

Added darktable.gui.select.selection_set_sensitive() to control the sensitivity of a lua registered selection.

Updated RELEASE_NOTES.md with lua changes so far.

Test script creates a button in the select images module, changes the sensitivity, then destroys the button.  There is a 1 second sleep between each change and output is to the console.

[selection.lua.txt](https://github.com/darktable-org/darktable/files/5938151/selection.lua.txt)

### CODE REVIEW

File: src/libs/select.c [hunk 1]
```diff
@@ -220,30 +226,141 @@ static int lua_register_selection(lua_State *L)
...
+  dt_lua_module_entry_push(L,"lib",self->plugin_name);
+  lua_getuservalue(L,-1);
+  lua_getfield(L,-1,"signal_handlers");
+  lua_pushstring(L,name);
+  lua_pushinteger(L,s);
+  lua_settable(L,-3);
+
   gtk_widget_show_all(self->widget);
+
+  if(tmp_name){
```
Comment (TurboGit) [inline src/libs/select.c:285]: { should be on next line. But in this case just:

`if(tmp_name) free(tmp_name);`

File: src/libs/select.c [hunk 2]
```diff
@@ -255,11 +372,29 @@ void init(struct dt_lib_module_t *self)
...
+  lua_pushcclosure(L, dt_lua_type_member_common, 1);
+  dt_lua_type_register_const_type(L, my_type, "set_sensitive");
+
   dt_lua_module_entry_push(L,"lib",self->plugin_name);
   lua_getuservalue(L,-1);
   lua_newtable(L);
   lua_setfield(L,-2,"callbacks");
   lua_pop(L,2);
+
+  dt_lua_module_entry_push(L,"lib",self->plugin_name);
```
Comment (TurboGit) [inline src/libs/select.c:393]: note that style is to have one space after a coma for readability. there is multiple instances.

File: src/libs/select.c [hunk 3]
```diff
@@ -220,30 +226,141 @@ static int lua_register_selection(lua_State *L)
...
+    name = luaL_checkstring(L,1);
+    key = luaL_checkstring(L,2);
+    func_pos = 3;
+  }
+  else
+  {
+    key = luaL_checkstring(L,1);
+    tmp_name = malloc(sizeof(key));
+    // replace spaces in name with underscores
+    for(int i = 0; key[i] != '\0'; i++)
```
Comment (TurboGit) [inline src/libs/select.c:246]: please wrap content of for loop with { and }.

**Review by TurboGit — CHANGES_REQUESTED** [review]
Some style comments. TIA.

File: src/libs/select.c (hunk 1 above)
Comment (wpferguson) [reply src/libs/select.c]: removed

File: src/libs/select.c (hunk 2 above)
Comment (wpferguson) [reply src/libs/select.c]: fixed

File: src/libs/select.c (hunk 3 above)
Comment (wpferguson) [reply src/libs/select.c]: removed

**Comment by wpferguson** [comment]
@TurboGit I owe you an apology.  I had written this, got it working, then saved it and moved on to the next thing.  When I pulled it back out, I just compiled it to make sure it was still working, but didn't look over the code (which was a mess).  I'm sorry I wasted your time by not doing my part properly.

**Review by TurboGit — APPROVED** [review]
Thanks !

**Pull request merged**